src/formatDate.js

```diff
@@ -8,7 +8,37 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.slice(-1)[0];
+  const toSeparator = toFormat.slice(-1)[0];
+  const dateElements = date.split(fromSeparator);
+
+  const dateMap = fromFormat.reduce((acc, format, index) => {
+    acc[format] = dateElements[index];
+
+    return acc;
+  }, {});
```

Comment:
The use of `reduce` here is appropriate, but you should ensure that the accumulator object keys are unique to the date parts. Since 'YYYY' and 'YY' could both appear in `fromFormat`, it's safer to check for uniqueness when assigning to `acc` to avoid potential overwrites.